Dockerfile

```diff
@@ -1,39 +1,65 @@
 # Base layer
 FROM ubuntu:22.04 AS baseline
+
 RUN apt-get update && apt-get upgrade -y --no-install-recommends \
-  && apt-get install -y python3 python3-dev python3-pip curl unzip gnupg --no-install-recommends \
+  && apt-get install -y \
+      python3 python3-dev python3-pip \
+      curl unzip gnupg lsb-release ca-certificates software-properties-common \
+      --no-install-recommends \
   && update-alternatives --install /usr/bin/python python /usr/bin/python3 1 \
   && update-alternatives --install /usr/bin/pip pip /usr/bin/pip3 1 \
   && apt-get clean && rm -rf /var/lib/apt/lists/*
 
-# Layers used for building/downloading/installing tools
+
+# Tool building layer
 FROM baseline AS tool_builder
+
 ARG HELM_VERSION=3.17.1
-ARG KUBECTL_VERSION=1.30.10
-ARG TERRAFORM_VERSION=1.10.5-*
+ARG KUBECTL_VERSION=1.32.7
 
-WORKDIR /build
+ARG TERRAFORM_VERSION=1.10.5
 
+WORKDIR /build
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
-RUN curl -fsSL https://apt.releases.hashicorp.com/gpg | apt-key add - \
-  && echo "deb [arch=amd64] https://apt.releases.hashicorp.com focal main" > /etc/apt/sources.list.d/tf.list \
-  && apt-get update \
-  && curl -sLO https://dl.k8s.io/release/v$KUBECTL_VERSION/bin/linux/amd64/kubectl && chmod 755 ./kubectl \
-  && curl -ksLO https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 && chmod 755 get-helm-3 \
-  && ./get-helm-3 --version v$HELM_VERSION --no-sudo \
-  && apt-get install -y terraform=$TERRAFORM_VERSION --no-install-recommends \
+
+# Install kubectl
+RUN curl -sLO https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl \
+  && chmod 755 ./kubectl
+
+# Install helm
+RUN curl -fsSL https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 -o get-helm-3 \
+  && chmod 755 get-helm-3 \
+  && ./get-helm-3 --version v${HELM_VERSION} --no-sudo
+
+# Install terraform (APT + fallback to binary)
+RUN set -e \
+  && curl -fsSL https://apt.releases.hashicorp.com/gpg | gpg --dearmor -o /usr/share/keyrings/hashicorp.gpg \
+  && echo "deb [signed-by=/usr/share/keyrings/hashicorp.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" > /etc/apt/sources.list.d/hashicorp.list \
+  && apt-get update || true \
+  && (apt-get install -y terraform=${TERRAFORM_VERSION} --no-install-recommends || \
+      (echo "APT install failed. Falling back to direct download..." && \
+       curl -fsSL https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip -o terraform.zip \
+       && unzip terraform.zip \
+       && mv terraform /usr/bin/terraform \
+       && chmod +x /usr/bin/terraform \
+       && rm terraform.zip)) \
```

Comment:
Was this a problem? Never saw where `apt install` failed? Guessing you all have seen this or this is part of this PR? The title says "Add kubernetes 1.33 support and upgrade Calico to 3.30.0"